polar2grid/writers/hdf5.py

```diff
@@ -0,0 +1,327 @@
+        """Delayed Geolocation Data write."""
+        msg = ("Adding geolocation 'longitude' and " "'latitude' datasets for grid %s", parent)
+        LOG.info(msg)
+        lon_data, lat_data = area_def.get_lonlats(chunks=chunks)
+
+        dtype = lon_data.dtype if dtype is None else dtype
+        data_shape = lon_data.shape
+
+        lon_grp = "{}/longitude".format(parent)
+        lat_grp = "{}/latitude".format(parent)
+
+        if append:
+            for var_name in [lon_grp, lat_grp]:
+                if var_name in fh:
+                    LOG.warning("Product %s already exists in hdf5 group, will delete existing dataset", var_name)
+                    del fh[var_name]
+
+        lon_dataset = FakeHDF5(fname, lon_grp)
+        lat_dataset = FakeHDF5(fname, lat_grp)
+        fh.create_dataset(lon_grp, shape=data_shape, dtype=dtype, compression=compression)
+        fh.create_dataset(lat_grp, shape=data_shape, dtype=dtype, compression=compression)
+
+        return [lon_data, lat_data], [lon_dataset, lat_dataset]
+
+    @staticmethod
+    def create_variable(
+        filename: str, hdf_fh, hdf_subgroup: str, data_arr: xr.DataArray, dtype: np.dtype, compression: bool
+    ):
+        """Create a hdf5 data variable and attributes for the variable."""
+        ds_attrs = data_arr.attrs
+
+        d_dtype = data_arr.dtype if dtype is None else dtype
+
+        if hdf_subgroup in hdf_fh:
+            LOG.warning("Product %s already in hdf5 group," "will delete existing dataset", hdf_subgroup)
+            del hdf_fh[hdf_subgroup]
+
+        try:
+            dset = hdf_fh.create_dataset(hdf_subgroup, shape=data_arr.shape, dtype=d_dtype, compression=compression)
+        except ValueError:
+            if os.path.isfile(filename):
+                os.remove(filename)
+            raise
+
+        dset.attrs["satellite"] = ds_attrs["platform_name"]
+        dset.attrs["instrument"] = ds_attrs["sensor"]
+        dset.attrs["begin_time"] = ds_attrs["start_time"].isoformat()
+        dset.attrs["end_time"] = ds_attrs["end_time"].isoformat()
+
+    def save_datasets(
+        self,
+        dataset: list[xr.DataArray],
+        filename=None,
+        dtype=None,
+        append=True,
+        compute=True,
+        chunks=None,
+        **kwargs,
+    ):
+        """Save hdf5 datasets."""
+        compression = kwargs.pop("compression", None) if "compression" in kwargs else None
+        if compression == "none":
+            compression = None
+
+        add_geolocation = kwargs.pop("add_geolocation") if "add_geolocation" in kwargs else False
+
+        # will this be written to one or multiple files?
+        output_names = []
+
+        for dataset_id in dataset:
+            file_attrs = self._output_file_kwargs(dataset_id, dtype)
+            out_filename = filename or self.get_filename(**file_attrs)
+            output_names.append(out_filename)
+
+        filename = output_names[0]
+        if not all_equal(output_names):
```

Comment:
```suggestion
        if not all(output_names[0] == output_name in output_name in output_names[1:]):
```

Potential "shorter" version that doesn't require additional imports.

Reply:
Or I guess since you did `filename = ` you could do `filename == output_name` in the loop.

Reply:
I changed the implementation of `all_equal` to use this technique. I had a hard time following how it was working in `all_equal`. I also changed the warning message to be more clear about why it is a warning.